test/models/autodeletable_test.rb

```diff
@@ -0,0 +1,112 @@
+                        line: @line,
+                        initial_call_date: 400.days.ago
+        @pt_d = create :archived_patient,
+                        line: @line,
+                        initial_call_date: 400.days.ago
+
+    end
+
+    # by default, no deletion: config is nil
+    describe 'without autodeletion' do
+        it 'should return nil deletion date' do
+            assert_nil Config.days_until_delete
+
+            assert_nil @pt_a.delete_date
+
+            assert_no_difference 'Patient.count' do
+                Patient.autodelete!
+            end
+
+            assert_no_difference 'ArchivedPatient.count' do
+                ArchivedPatient.autodelete!
+            end
+        end
+    end
+
+    describe 'with autodeletion' do
+        before do
+            c = Config.find_or_create_by(config_key: 'days_until_delete')
+            c.config_value = { options: ["100"] }
+            c.save!
+        end
+
+        it 'should only delete patients outside the window' do
+            assert_difference 'Patient.count', -1 do
+                Patient.autodelete!
+            end
+
+            # specifically, that should be pt_c
+            refute_includes Patient.all, @pt_c
+
+            assert_difference 'ArchivedPatient.count', -1 do
+                ArchivedPatient.autodelete!
+            end
+        end
+    end
+
+    describe 'edge cases' do
+        describe 'config 0 days' do
+            before do
+                c = Config.find_or_create_by(config_key: 'days_until_delete')
+                c.config_value = { options: ["0"] }
+                c.save!
+            end
+
+            it 'if config is 0, delete all patients now' do
+                Timecop.freeze(1.hour.ago) do
+                    Patient.autodelete!
+                    # delete doesn't occur because now hasn't happened yet ;)
+                    assert_equal 1, Patient.count
+                end
+
+                Timecop.freeze(1.day.after) do
+                    # later, now has happened
+                    Patient.autodelete!
+                    assert_equal 0, Patient.count
+                end
+            end
+        end
+        
+        describe 'nonzero config on the edge' do
+            before do
+                c = Config.find_or_create_by(config_key: 'days_until_delete')
+                c.config_value = { options: ["10"] }
+                c.save!
+            end
+
+            it 'deletes pt_a on the right day' do
+                Timecop.freeze(1.day.ago) do
+                    Patient.autodelete!
+                    assert_includes Patient.all, @pt_a
+                end
+
+                # present day - doesn't delete because <, not <=
+                Patient.autodelete!
+                assert_includes Patient.all, @pt_a
+
+                Timecop.freeze(1.day.after) do
+                    Patient.autodelete!
+                    refute_includes Patient.all, @pt_a
+                end       
```

Comment:
ok this is a lil weird, because i'm using `<` with `today`, so it does a _date_-, not _time_-based comparison. I don't particularly have an issue with this -- I don't think funds will care _exactly_ which day a patient deletes. but let me know and I can make it a bit more intuitive.